TypeScript auto configure

# Bug report

## Describe the bug

Following the [docs](https://nextjs.org/docs#typescript), when an empty `tsconfig.json` file is in the root running `next dev` throws the following error after installing the typescript packages:

```
error TS18003: No inputs were found in config file 'tsconfig.json'. Specified 'include' paths were '["**/*"]' and 'exclude' paths were '[]'.
```

## To Reproduce

```
touch tsconfig.json
npm run dev
# Install the packages as recommended
npm run dev
```

## Expected behavior

The `tsconfig.json` file should be auto-configured with default values as per the docs.

## System information

- macOS 10.14.5
- Next 9.0.3
- @types/node: 12.7.1
- @types/react: 16.9.1
- typescript: 3.5.3

--- a/packages/next/lib/verifyTypeScriptSetup.ts
+++ b/packages/next/lib/verifyTypeScriptSetup.ts
@@ -192,6 +192,14 @@ export async function verifyTypeScriptSetup(dir: string): Promise<void> {
       path.dirname(tsConfigPath)
     )
 
+    if (result.errors) {
+      result.errors = result.errors.filter(
+        ({ code }) =>
+          // No inputs were found in config file
+          code !== 18003
+      )
+    }
+
     if (result.errors && result.errors.length) {
       throw new Error(
         ts.formatDiagnostic(result.errors[0], formatDiagnosticHost)

--- a/test/integration/tsconfig-verifier/test/index.test.js
+++ b/test/integration/tsconfig-verifier/test/index.test.js
@@ -1,7 +1,14 @@
 /* eslint-env jest */
 /* global jasmine */
 import path from 'path'
-import { exists, remove, readFile, readJSON, writeJSON } from 'fs-extra'
+import {
+  exists,
+  remove,
+  readFile,
+  readJSON,
+  writeJSON,
+  createFile
+} from 'fs-extra'
 
 import { launchApp, findPort, killApp, renderViaHTTP } from 'next-test-utils'
 
@@ -25,6 +32,28 @@ describe('Fork ts checker plugin', () => {
 
   it('Creates a default tsconfig.json when one is missing', async () => {
     expect(await exists(tsConfig)).toBe(true)
+
+    const tsConfigContent = await readFile(tsConfig)
+    let parsedTsConfig
+    expect(() => {
+      parsedTsConfig = JSON.parse(tsConfigContent)
+    }).not.toThrow()
+
+    expect(parsedTsConfig.exclude[0]).toBe('node_modules')
+  })
+
+  it('Works with an empty tsconfig.json (docs)', async () => {
+    await killApp(app)
+
+    await remove(tsConfig)
+    await new Promise(resolve => setTimeout(resolve, 500))
+    expect(await exists(tsConfig)).toBe(false)
+    await createFile(tsConfig)
+    await new Promise(resolve => setTimeout(resolve, 500))
+    expect(await readFile(tsConfig, 'utf8')).toBe('')
+
+    appPort = await findPort()
+    app = await launchApp(appDir, appPort)
 
     const tsConfigContent = await readFile(tsConfig)
     let parsedTsConfig

